src/Autocomplete.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from './types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [noSuggestions, setNoSuggestions] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      const trimmedInput = inputValue.trim();
+
+      if (trimmedInput === '') {
+        setSuggestions(people);
+        setNoSuggestions(false);
+
+        return;
+      }
+
+      const filteredPeople = people.filter(person =>
+        person.name.toLowerCase().includes(trimmedInput.toLowerCase()),
+      );
+
+      setSuggestions(filteredPeople);
+      setNoSuggestions(filteredPeople.length === 0);
+    }, delay);
+
+    return () => clearTimeout(timeoutId);
+  }, [inputValue, people, delay]);
+
+  const handlePersonSelect = (person: Person) => {
+    setInputValue(person.name);
+    setSelectedPerson(person);
+    setSuggestions([]);
+    onSelected(person);
```

Comment:
The checklist mentions that you should follow specific naming conventions for methods. Your method `handlePersonSelect` is correctly named according to the conventions. However, `handleInputValueChange` could be renamed to `handleInputChange` to be more concise and still clear.